pkg/datapath/loader/loader.go

```diff
@@ -97,6 +98,40 @@ func CompileAndLoad(ctx context.Context, ep endpoint) error {
 	return compileAndLoad(ctx, ep, &dirs)
 }
 
+// CompileOrLoad loads the BPF datapath programs for the specified endpoint.
+//
+// In contrast with CompileAndLoad(), it attempts to find a pre-compiled
+// template datapath object to use, to avoid a costly compile operation.
+// Only if there is no existing template that has the same configuration
+// parameters as the specified endpoint, this function will compile a new
+// template for this configuration.
+//
+// This function will block if the cache does not contain an entry for the
+// same EndpointConfiguration and multiple goroutines attempt to concurrently
+// CompileOrLoad with the same configuration parameters. When the first
+// goroutine completes compilation of the template, all other CompileOrLoad
+// invocations will be released.
+func CompileOrLoad(ctx context.Context, ep endpoint) error {
+	templatePath, _, err := templateCache.fetchOrCompile(ctx, ep)
+	if err != nil {
+		return err
+	}
+
+	template, err := elf.Open(templatePath)
+	if err != nil {
+		return err
+	}
+	defer template.Close()
+
+	dstPath := path.Join(ep.StateDir(), endpointObj)
+	opts, strings := ELFSubstitutions(ep)
+	if err = template.Write(dstPath, opts, strings); err != nil {
+		return err
```

Comment:
Does the error returned contain the `dstPath`?

Reply:
Yes, the `pkg/elf` library puts the path into the error.